### ISSUE TITLE
Unable to Connect to NiFi Host with Authentication

### ISSUE THREAD

**Issue by guptasakshi01** [Chaffelson/nipyapi#190]
* Nipyapi version: 0.14.2
* Python version: 3.6
* Operating System: Windows

### Description

Trying to Connect to NiFi host and list all templates present in that environment

### What I Did

import nipyapi
nipyapi.config.registry_config.verify_ssl = False
nipyapi.config.nifi_config.verify_ssl = False
nipyapi.config.nifi_config.host = host
nipyapi.config.nifi_config.username=username
nipyapi.config.nifi_config.password=password
print(nipyapi.templates.list_all_templates(native=True))

### Traceback

nipyapi.nifi.rest.ApiException: (401)
Reason: Unauthorized
HTTP response headers: HTTPHeaderDict({'Date': 'Thu, 09 Apr 2020 06:27:43 GMT', 'X-Frame-Options': 'SAMEORIGIN', 'Content-Security-Policy': "frame-ancestors 'self'", 'X-XSS-Protection': '1; mode=block', 'Strict-Transport-Security': 'max-age=31540000', 'Server': 'Jetty(9.4.11.v20180605)', 'X-ProxiedEntitiesAccepted': 'true', 'Vary': 'Accept-Encoding', 'Content-Type': 'text/plain', 'Content-Length': '73'})
HTTP response body: Unknown user with identity 'anonymous'. Contact the system administrator

Can someone help how can we handle Authorization while connecting to NiFi Host ?

**Comment by Chaffelson** [comment]
Is your environment just using basic Auth, not token ?

**Comment by guptasakshi01** [comment]
It Supports Both.

**Comment by guptasakshi01** [comment]
@Chaffelson 
```
nipyapi.security.service_login(service='nifi',username='<username>',password='<password>',bool_response=False)
templates = nipyapi.templates.list_all_templates(native=True)
```
Above code is working as expected but for nipyapi verion 0.14.1 but not working for 0.14.2(latest version)

Is this functionality going to be deprecated in further release, or is this a bug to be fixed in 0.14.2 ?

**Comment by Chaffelson** [comment]
Looks like I've introduced a bug.
I had thought that adding basic Auth would not break token, but it seems
more testing is needed. Please pin to 0.14.1 for now with my apologies

**Comment by guptasakshi01** [comment]
@Chaffelson 
No Issues
Thanks for writing a great wrapper, its really of much help :)
Closing this issue, as it works with version 0.14.1

**Comment by Chaffelson** [comment]
Reopening to use this issue to fix the bug in 0.14.2
Found that the way the mustache templates generate the client, basicAuth will clobber tokenAuth if it is enabled. Previously we did not allow basicAuth in most functions so the effect was not noticed.
Am testing a simple configuration control for which Auth Type you want enabled. I guess that most users are like me and set the auth method per-session based on the endpoint I'm connecting to
